tree/ntuple/v7/inc/ROOT/RNTupleZip.hxx

```diff
@@ -39,23 +40,31 @@ namespace Detail {
 // clang-format on
 class RNTupleCompressor {
 private:
-   using Buffer_t = std::array<unsigned char, kMAXZIPBUF>;
-   std::unique_ptr<Buffer_t> fZipBuffer;
-
+   std::vector<unsigned char> fZipBuffer;
+   std::size_t GetBlockSize() const { return fZipBuffer.size(); }
 public:
    /// Data might be overwritten, if a zipped block in the middle of a large input data stream
    /// turns out to be uncompressible
    using Writer_t = std::function<void(const void *buffer, size_t nbytes, size_t offset)>;
    static constexpr size_t kMaxSingleBlock = kMAXZIPBUF;
 
-   RNTupleCompressor() : fZipBuffer(std::unique_ptr<Buffer_t>(new Buffer_t())) {}
+   /// Make a compressor with the default scratch buffer size, kMAXZIPBUF (16MB).
+   RNTupleCompressor() : fZipBuffer(std::vector<unsigned char>(kMaxSingleBlock, 0)) {}
+   /// Make a compressor with a specific scratch buffer size. Sizes larger than
+   /// kMAXZIPBUF will be set to kMAXZIPBUF.
+   explicit RNTupleCompressor(std::size_t bufSize) {
+      if (bufSize > kMaxSingleBlock) {
+         bufSize = kMaxSingleBlock;
```

Comment:
Is this safe? We could also error out on this condition...

Reply:
Revised to work without the RNTupleCompessor changes